curp/src/error.rs

```diff
@@ -101,17 +80,8 @@ impl TryFrom<PbProposeError> for ProposeError {
     #[inline]
     fn try_from(err: PbProposeError) -> Result<ProposeError, Self::Error> {
         Ok(match err {
-            PbProposeError::Timeout(_) => ProposeError::Timeout,
-            PbProposeError::NotLeader(_) => ProposeError::NotLeader,
-            PbProposeError::Shutdown(_) => ProposeError::Shutdown,
             PbProposeError::KeyConflict(_) => ProposeError::KeyConflict,
             PbProposeError::Duplicated(_) => ProposeError::Duplicated,
-            PbProposeError::WaitSyncError(e) => ProposeError::SyncedError(
-                e.wait_sync_error
-                    .ok_or(PbSerializeError::EmptyField)?
-                    .into(),
-            ),
-            PbProposeError::EncodeError(s) => ProposeError::EncodeError(s),
         })
     }
```

Comment:
If this error does not contain extra information internally, Please use enum instead of oneof in proto.
And `curp-error.proto` only has one message, Should we merge this file into `curp-command.proto`?